init/pipeline.py

```diff
@@ -1,43 +1,54 @@
 import dlt
-from dlt.sources.helpers import requests
 
+from dlt.sources.helpers.rest_client import paginate
+from dlt.sources.helpers.rest_client.auth import BearerTokenAuth
+from dlt.sources.helpers.rest_client.paginators import HeaderLinkPaginator
 
-@dlt.source
-def source(api_secret_key=dlt.secrets.value):
-    return resource(api_secret_key)
-
-
-def _create_auth_headers(api_secret_key):
-    """Constructs Bearer type authorization header which is the most common authorization method"""
-    headers = {"Authorization": f"Bearer {api_secret_key}"}
-    return headers
+# This pipeline demonstrates how to build a simple REST client for interacting with GitHub's API.
+# It showcases the use of authentication via bearer tokens and pagination for navigating through
+# GitHub issues and pull requests within a repository.
 
+# Note: Ensure the API key (Bearer token) is set up correctly in secrets or environment variables.
 
-@dlt.resource(write_disposition="append")
-def resource(api_secret_key=dlt.secrets.value):
-    headers = _create_auth_headers(api_secret_key)
 
-    # check if authentication headers look fine
-    print(headers)
+@dlt.source
+def source(
+    api_url: str = dlt.config.value,
+    api_secret_key: str = dlt.secrets.value,
+    repository: str = dlt.config.value,
+):
+    return my_repo_pulls(api_url, api_secret_key, repository)
 
-    # make an api call here
-    # response = requests.get(url, headers=headers, params=params)
-    # response.raise_for_status()
-    # yield response.json()
 
-    # test data for loading validation, delete it once you yield actual data
-    test_data = [{"id": 0}, {"id": 1}]
-    yield test_data
+@dlt.resource(write_disposition="append")
+def my_repo_pulls(
+    api_url: str = dlt.config.value,
+    api_secret_key: str = dlt.secrets.value,
+    repository: str = dlt.config.value,
+):
+    # repository url should be in the format `OWNER/REPO`
+
+    # paginate pull requests and yield every page
+    url = f"{api_url}/repos/{repository}/pulls"
+    for page in paginate(
+        url,
+        auth=BearerTokenAuth(api_secret_key),
+        paginator=HeaderLinkPaginator(),
```

Comment:
Let's add a comment with a not that other paginators are available for different strategies and link https://dlthub.com/devel/general-usage/http/rest-client#paginators